packages/main/src/RPA/JavaAccessBridge.py

```diff
@@ -0,0 +1,520 @@
+        top = int(self.display_scale_factor * (element.aci.y))
+        width = int(self.display_scale_factor * (element.aci.width))
+        height = int(self.display_scale_factor * (element.aci.height))
+        right = left + width
+        bottom = top + height
+        return left, top, right, bottom
+
+    def _get_region_locator(self, element):
+        left, top, right, bottom = self._get_scaled_coordinates(element)
+        return Desktop().define_region(left, top, right, bottom)
+
+    @keyword
+    def click_element(
+        self,
+        target: LocatorType,
+        index: int = 0,
+        action: bool = True,
+        timeout: int = 10,
+    ):
+        """Click element
+
+        :param target: element to click
+        :param index: target element if multiple are returned
+        :param action: call click action on element (default), or use coordinates
+        :param timeout: timeout in seconds to find element
+        """
+        if isinstance(target, str):
+            interval = float(0.2)
+            end_time = time.time() + float(timeout)
+            while time.time() <= end_time:
+                start = time.time()
+                elements = self._find_elements(target)
+                if len(elements) > 0:
+                    break
+                duration = time.time() - start
+                if duration < interval:
+                    time.sleep(interval - duration)
+
+            if len(elements) < (index + 1):
+                raise ElementNotFound(
+                    "Locator '%s' matched only %s elements" % (target, len(elements))
+                )
+            matching = elements[index]
+        else:
+            matching = target
+        try:
+            if action:
+                self.logger.info("Element click action")
+                matching.do_action("click")
+            else:
+                self._click_element_middle(matching)
+        except NotImplementedError:
+            self._click_element_middle(matching)
+
+    @keyword
+    def call_element_action(self, locator: str, action: str):
+        """Call element action
+
+        :param locator: target element
+        :param action: name of the element action to call
+        """
+        elements = self._find_elements(locator)
+        if len(elements) != 1:
+            raise ElementNotFound("Locator %s did not match a unique element" % locator)
+        matching = elements[0]
+        self.logger.info("Element '%s' action", action)
+        matching.do_action(action)
+
+    def _click_element_middle(self, element):
+        self.logger.info("Element click coordinates")
+        left, top, right, bottom = self._get_scaled_coordinates(element)
+        if left == -1 or top == -1:
+            raise AttributeError("Can't click on negative coordinates")
+        middle_x = int((left + right) / 2)
+        middle_y = int((top + bottom) / 2)
+        point = f"point:{middle_x},{middle_y}"
+        Desktop().click(point)
```

Comment:
The `Desktop` library will accept geometry primitives defined in `RPA.core.geometry`, i.e. `Region` and `Point` instances. No need to convert things into a string first. You can probably do just something like this:
```
region = Region.from_size(element.left, element.top, element.width, element.height)
region.scale(self.scale_factor)
Desktop().click(region.center)
```

Reply:
Seems like nice way of using Region. I will add this as an library TODO ... no time to change and retest everything this week.